app/models/order.rb

```diff
@@ -0,0 +1,31 @@
+class Order < ApplicationRecord
+  enum :status, {'In Progress': 0, 'Placed': 1, 'Delivered': 2, 'Cancelled': 3}
+  
+  belongs_to :user
+  belongs_to :address, optional: true
+  has_many :payments
+  has_many :line_items, dependent: :destroy
+  has_many :deals, through: :line_items
+
+  scope :placed_orders, ->{ includes(:address, :payments).where.not(status: 'In Progress').where(payments: { status: 'Successful' }) }
+  scope :users_placed_orders, ->(user_id){ where(user_id: user_id).placed_orders }
+  
+  def total_tax
+    line_items.sum(&:tax_on_deal)
+  end
+
+  def order_total
+    total = line_items.sum(&:total_price)
+    order_count = user.orders.size
+
+    if order_count > 3
```

Comment:
on order there should be following fields
total_in_cents - Sum of line items' total
discount in cents - Sum of line items' discounts
loyalty_discount_in_cents - as per the condition given in the doc
net_in_cents - total - discount - loyalty

Reply:
No need to calculate it every time

Reply:
done